### ISSUE TITLE
Crash when downloading pdf without content license

### ISSUE THREAD

**Issue by pixil98** [rmcrackan/Libation#358]
**Describe the bug**
When testing out my docker image I attempted to download [The Count of Monte Cristo](https://www.audible.com/pd/The-Count-of-Monte-Cristo-Audiobook/B005GFQ5WQ). I believe this used to be an Audible Plus book so I'd expect it to fail to download.

You can see from the logs that the book did fail to download.
```
DownloadDecryptBook Begin: 05/11/2022 [B005GFQ5WQ] The Count of Monte Cristo
DownloadDecryptBook Completed: 05/11/2022 [B005GFQ5WQ] The Count of Monte Cristo
DownloadPdf Begin: 05/11/2022 [B005GFQ5WQ] The Count of Monte Cristo
Error processing book. Skipping. This book will be tried again on next attempt. For options of skipping or marking as error, retry with main Libation app.. See log for more details.
```

The PDF however, caused a core dump on it's failure:
```
Unhandled exception. AudibleApi.ValidationErrorException: Content License denied for asin: [B005GFQ5WQ]
   at AudibleApi.Api.GetDownloadLicenseAsync(String asin, DownloadQuality quality) in /home/runner/work/AudibleApi/AudibleApi/AudibleApi/Api.Download.cs:line 123
   at AudibleApi.Api.GetPdfDownloadLinkAsync(String asin) in /home/runner/work/AudibleApi/AudibleApi/AudibleApi/Api.Download.cs:line 141
   at FileLiberator.DownloadPdf.downloadPdfAsync(LibraryBook libraryBook, String proposedDownloadFilePath) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\Source\FileLiberator\DownloadPdf.cs:line 61
   at FileLiberator.DownloadPdf.ProcessAsync(LibraryBook libraryBook) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\Source\FileLiberator\DownloadPdf.cs:line 30
   at FileLiberator.Processable.TryProcessAsync(LibraryBook libraryBook) in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\Source\FileLiberator\Processable.cs:line 61
   at LibationCli.LiberateOptions.<>c__DisplayClass5_0.<<CreateBackupBook>g__onDownloadDecryptBookCompleted|0>d.MoveNext() in C:\Dropbox\DinahsFolder\coding\_NET\Visual Studio 2022\Libation\Source\LibationCli\Options\LiberateOptions.cs:line 30
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
   at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
./liberate.sh: line 11:    36 Aborted                 (core dumped) /libation/LibationCli liberate
```

Interestingly it says that it finished downloading the PDF right before the exception.

**To Reproduce**

1. Attempt to download a pdf you don't have the content license for using the cli.

**Expected behavior**
I'd expect it to gracefully not download the pdf

**Platform**
Linux

**Comment by Mbucari** [comment]
@rmcrackan This is happening because `DownloadPdf.TryProcessAsync` is being called inside an async event handler. When DownloadDecryptBook.Completed fires, it runs the event handler on a new thread and so the handler is outside the scope of the OptionsBase.Run exception handler.

https://github.com/rmcrackan/Libation/blob/1510a86579a7bb19a8d372d355674955c2b5668e/Source/LibationCli/Options/LiberateOptions.cs#L28-L31

There are a dozen ways to fix this, so I'll let you choose a solution you like.

**Comment by rmcrackan** [comment]
Code is complete. The next version will contain the fix. I'm currently fighting through a publishing issue first.

**Comment by rmcrackan** [comment]
I just released [v8.4.2](https://github.com/rmcrackan/Libation/releases/tag/v8.4.2) to address your issue. Please try it and let me know if this fixes your error.

This will not get picked up with auto-update since it's technically a "pre-release". I'm going on vacation soon and I don't want to risk putting out a bad full release then people are stuck with it until I return to angry emails.

**Comment by rmcrackan** [comment]
I changed 8.4.2 to a full release. It should fix your issue. If not, please reply here and we can re-open the ticket.
